pkg/task/task.go

```diff
@@ -0,0 +1,197 @@
+/*
```

Comment:
I added a commit [here](https://github.com/jcooklin/snap/tree/fix_4_issue_950) that removes the task pkg and moves the logic to core/task.go and core/schedule.go just to look at what a possible alternative looks like.